cogito/protocol_test.go

```diff
@@ -89,9 +104,45 @@ func TestSourceValidationFailure(t *testing.T) {
 
 	testCases := []testCase{
 		{
-			name:    "missing mandatory git source keys",
+			name:    "missing access token or github_app",
```

Comment:
Ah. I got confused here. I think that a clearer test name is "access_token and github_app both missing".